websocket.go

```diff
@@ -24,28 +24,32 @@ var upgrader = websocket.Upgrader{
 }
 
 //reader listens for new messages being sent to our WS endpoint
-func reader(conn *websocket.Conn, errCh chan error) {
+func reader(hub *Hub, conn *websocket.Conn) {
 	for {
 		// read in message
 		_, p, err := conn.ReadMessage()
 		if err != nil {
 			log.Warning(err)
-			errCh <- err
+			delete(hub.clients, conn)
+			log.Println(hub.clients)
 			return
 		}
 		log.Println(string(p))
 	}
 }
 
-func writer(conn *websocket.Conn) {
-	err := conn.WriteMessage(1, []byte(`{"message": "Hi Client!"}`))
-	if err != nil {
-		log.Println(err)
+func writer(conn *websocket.Conn, msgChan chan []byte) {
+	for {
+		msg := <-msgChan
+		if err := conn.WriteMessage(1, msg); err != nil {
+			log.Warning(err)
+			return
```

Comment:
have some problems here. Sometimes it can be, that connection is deleted, but data try to be sent through closed WebSocket, then this error occurred. Tried to fix it with adding chan in reader that provide about client deletion and send info to the same chan in writer in order to close connection, but the problem wasn't solved. Need your help in this case.
<img width="835" alt="Снимок экрана 2020-07-22 в 00 38 15" src="https://user-images.githubusercontent.com/39093991/88109707-a7061680-cbb3-11ea-9ff8-ef6adf9bb475.png">

Reply:
bug was fixed with the last commit